buildSrc/src/main/java/com/osacky/flank/gradle/YamlWriter.kt

```diff
@@ -12,30 +12,34 @@ internal class YamlWriter {
       check(base.serviceAccountCredentials.isPresent) { "ServiceAccountCredentials in fladle extension not set. https://github.com/runningcode/fladle#serviceaccountcredentials" }
     }
     check(base.debugApk.isPresent) { "debugApk must be specified" }
-    check(base.instrumentationApk.isPresent xor !base.roboScript.orNull.isNullOrBlank()) {
-      """
+    if (config.sanityRobo.get() == false) {
+      check(config.instrumentationApk.isPresent xor !config.roboScript.orNull.isNullOrBlank()) {
+        """
      Either instrumentationApk file or roboScript file must be specified but not both.
-     instrumentationApk=${base.instrumentationApk.orNull}
-     roboScript=${base.roboScript.orNull}
-      """.trimIndent()
+     instrumentationApk=${config.instrumentationApk.orNull}
+     roboScript=${config.roboScript.orNull}
+        """.trimIndent()
+      }
     }
 
-    val additionalProperties = writeAdditionalProperties(config)
-    val flankProperties = writeFlankProperties(config)
+    val shouldPrintTestAndRobo = config.sanityRobo.get().not()
+    val additionalProperties = writeAdditionalProperties(config, shouldPrintTestAndRobo)
+    val flankProperties = writeFlankProperties(config, shouldPrintTestAndRobo)
 
     return buildString {
       appendln("gcloud:")
-      appendln("  app: ${base.debugApk.get()}")
-      if (base.instrumentationApk.isPresent) {
-        appendln("  test: ${base.instrumentationApk.get()}")
+      appendln("  app: ${config.debugApk.get()}")
+      // We don't want to print instrumentation apks if sanityRobo == true
```

Comment:
shouldn't we throw an error here if sanityRobo is true?

Reply:
Sanity run validation if the user sets `instrumentationApk` explicitly and this occurs before `findDebugAndInstrumentationApk` (autodetection ) is invoked (in `FlankPluginDelegate`). The autodetection feature sets app apk and test apk only for `base` config and then those values are used for inner (or would should I name those additional ?) configs.

So actually this is applicable only for case when `base` config is intended to be sanity robo run and the rest 'normal'

Logic could be simplified if we would agree that `base` config couldn't be sanity robo